### PR TITLE
Added functions for setting the c/c++ language standard on projects

### PR DESCRIPTION
**Pull request by zoejbare**

### CODE REVIEW

File: csbuild/tools/project_generators/visual_studio/internal/__init__.py [hunk 1]
```diff
@@ -1107,28 +1119,34 @@ def _writeFiltersVcxProj(outputRootPath, project):
...
+					continue
 
-			platformHandler = PLATFORM_HANDLERS[buildSpec]
-			platformHandler.WriteUserDebugPropertyGroup(rootXmlNode, project, buildSpec, _getVsConfigName(buildSpec))
+				platformHandler = PLATFORM_HANDLERS[buildSpec]
+				platformHandler.WriteUserDebugPropertyGroup(rootXmlNode, project, buildSpec, _getVsConfigName(buildSpec))
+
+	else:
+		# No output file needed.
+		rootXmlNode = None
```
[the remark this answers is not in the document]
Comment (zoejbare) [reply csbuild/tools/project_generators/visual_studio/internal/__init__.py:1143]: It's needed by VsFileProxy for logging.  If the root node is `None`, it won't save anything to disk, but it outputs a log message that it's skipping that file.  It felt better to me at the time to put it in there.

[the remark this answers is not in the document]
Comment (ShadauxCat) [reply csbuild/tools/project_generators/visual_studio/internal/__init__.py:1137]: That wasn't really what I was asking. You have:

```python
if foo:
    rootXmlNode = (something)
else:
    rootXmlNode = None
    
_saveXmlFile(outputFilePath, rootXmlNode)
```

Why not just


```python
if foo:
    rootXmlNode = (something)
    _saveXmlFile(outputFilePath, rootXmlNode)
else:
    rootXmlNode = None
```

That way you're not doing extra work to check details of rootXmlNode within _saveXmlFile().

If _saveXmlFile() is doing more than just writing the file to disk, I would argue it's poorly named - it does more than it says it does. If that's the case, I'd recommend a more descriptive name.

File: csbuild/tools/cpp_compilers/cpp_compiler_base.py [hunk 2]
```diff
@@ -47,7 +49,14 @@ def _ignore(_):
 	pass
 
 @MetaClass(ABCMeta)
-class CppCompilerBase(HasDebugLevel, HasDebugRuntime, HasDefines, HasIncludeDirectories, HasOptimizationLevel, HasStaticRuntime):
+class CppCompilerBase(HasDebugLevel,
```
[the remark this answers is not in the document]
Comment (zoejbare) [reply csbuild/tools/cpp_compilers/cpp_compiler_base.py]: Done

File: csbuild/tools/project_generators/visual_studio/internal/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (zoejbare) [reply csbuild/tools/project_generators/visual_studio/internal/__init__.py]: Rearranged to remove the `None` rootXmlNode.

**Review by ShadauxCat — DISMISSED** [review]

**Review by ShadauxCat — APPROVED** [review]

**Pull request merged**